ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/util/VersionUtil.java

```diff
@@ -120,4 +120,14 @@ public static String computeInitialPluginVersion(String version) {
 		return version;
 	}
 
+	public static String computeInitialRequirementVersionRange(String version) {
+		if (version != null && VersionUtil.validateVersion(version).isOK()) {
+			Version pvi = Version.parseVersion(version);
+			return new VersionRange(
+					"[" + pvi.getMajor() + "." + pvi.getMinor() + "," + (pvi.getMajor() + 1) + ")") //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$//$NON-NLS-4$
+							.toString();
+		}
+		return ""; //$NON-NLS-1$
+	}
```

Comment:
Recently I added a method that does this to `ManifestUtils`:
https://github.com/eclipse-pde/eclipse.pde/blob/6a28c28a41ebf2634f1ef960283cfb0014491d76/ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/util/ManifestUtils.java#L402-L410

Admittedly this method would probably better be moved to this class and probably be used instead of the existing method. That would hopefully add proper version ranges in more places.
I can try to have a look at unifying these as soon as possible. Or if you want to have a look at it, it would be more than welcome.